채팅방 이동 주요 기능

youzysu opened the pull request:
## Issues
- #76 

## What is this PR? 👓
- 채팅방 이동 주요 기능
- 기타 리팩토링: useToast hook, 상품 가격 컴포넌트 분리

## To Reviewer
- 아직 백엔드 개발 서버에 API가 미완성이라 확인 못한 부분이 있어서 내일 확인할 예정이라 draft로 둘게요.

youzysu committed:
refactor: useToast hook 내부 toast 함수 useCallback 사용

youzysu committed:
refactor: ChatBar input text 여부로 전송 disabled 여부 추가

youzysu committed:
feat: 채팅방 나가기 기능 및 채팅방 생성 API 수정사항 반영

youzysu committed:
refactor: 상품 가격 나타내는 컴포넌트 분리

youzysu committed:
feat: 제품 상세 페이지 채팅 이동 버튼 컴포넌트 분리

- 구매자/판매자에 따라 컴포넌트 렌더링

youzysu committed:
feat: 채팅 상세 페이지에서 채팅방 진입 주요 기능

- 채팅룸 존재 여부 조회 & 이전 채팅 데이터 요청
- 채팅룸 삭제 이후 채팅 목록 재조회 연결
- 채팅룸 id 타입 전반 반영 (number -> string)

youzysu committed:
fix: 채팅목록페이지 router 에러 수정

youzysu committed:
style: 코드 컨벤션을 위한 단순 수정

aaaz425 commented on fe/src/router/router.tsx:
```diff
@@ -36,10 +36,12 @@ export const router = createBrowserRouter(
-            path={`${ROUTE_PATH.chat}/:productId`}
-            element={<ChatList />}
-          />
+          <Route path={ROUTE_PATH.chat} element={<ChatList />}>
+            <Route
+              path={`${ROUTE_PATH.chat}/:productId`}
+              element={<ChatList />}
+            />
+          </Route>
```
path에 optional 붙이면 된다고 합니다 [참고 링크](https://stackoverflow.com/questions/35604617/react-router-with-optional-path-parameter)
```javascript
<Route
  path={`${ROUTE_PATH.chat}/:productId?`}
  element={<ChatList />}
/>
```

v6에서 지원하지 않는다고 하는데...
[참고링크](https://github.com/remix-run/react-router/issues/7285)
[참고링크2](https://github.com/remix-run/react-router/discussions/9208)

결국 다시 지원하는 것 같아요
[참고링크](https://github.com/remix-run/react-router/issues/9546)

제 로컬에서는 optional 정상 작동 됩니다!

youzysu committed:
refactor: 채팅 API 변경 사항 반영

youzysu committed:
refactor: STOMP Client hook 분리

youzysu committed:
refactor: 채팅 한글 합성 문제 해결 & autoFocus 옵션 추가

youzysu committed:
feat: message sub, pub 시 header에 token 추가

youzysu committed:
fix: 웹소켓 반복 재연결되는 문제 해결

- useWebsocket hook 내부 useEffect dependencies로 추가해야 하는 appendNewChat 함수 참조값 유지를 위해 useCallback 사용

youzysu committed:
fix: 채팅방 생성 후 ws 연결 안되는 문제 해결

- 현재 useWebSocket custom hook 실행 관련 문제(roomId 변경 문제)
- 분리해둔 로직 우선 임시로 다시 합쳐둠

youzysu committed:
refactor: 채팅방 상품 타입 개선 & 전반 반영

- id -> productId로 채팅 목록 조회 API response와 통일하기 위함

youzysu committed:
Merge branch 'dev' of https://github.com/masters2023-project-team05-second-hand/second-hand-max-fe into feat/76-chat-from-product-detail
